presto-docs/src/main/sphinx/release/release-341.md

```diff
@@ -0,0 +1,143 @@
+* Fix serialization of NULL values in structural types for old Presto clients ({issue}`4778`)
+* Fix failure when aggregation query contains duplicate expressions. ({issue}`4872`)
+* Fix compiler failure when querying long timestamp type. ({issue}`4824`)
+* Change timestamp operations to match the SQL specification. The value of a `TIMESTAMP` type is not affected by the session 
+  time zone. ({issue}`4799`)
+* Fix parsing failure of timestamps due to daylight saving changes. ({issue}`4799`)
+* Remove the `deprecated.legacy-timestamp` configuration property and the `legacy_timestamp` session property. ({issue}`4799`)
+* Preserve precision when applying `AT TIME ZONE` to values of type `TIMESTAMP`. ({issue}`4866`)
+* Fix failure when calling `EXTRACT` with `TIMEZONE_HOUR` and `TIMEZONE_MINUTE` for `TIMESTAMP WITH TIME ZONE` type. ({issue}`4867`)
+* Fix query deadlock for connectors that wait for dynamic filters. ({issue}`4946`)
+* Add `optimizer_rule_stats` system table. ({issue}`4659`)
+* Fix failure when `TIME` or `TIMESTAMP` subtraction returns a negative value. ({issue}`4847`)
+* Fix failure when duplicate expressions appear in `DISTINCT` clause. ({issue}`4787`)
+* Fix failure for join queries during spilling or when available memory is low. ({issue}`4994`)
+
+## Security Changes
+
+* Add Salesforce password authentication. ({issue}`4372`)
+* Allow to configure refresh period for system level access control with environment variables. ({issue}`4854`)
+* Only request HTTPS client certificate when certificate authentication is enabled. ({issue}`4804`)
+* Allow to lowercase/uppercase matched user name in user mapping. ({issue}`4736`)
+
+## Web UI
+
+* Fix displaying physical input read time in detailed query view. ({issue}`4962`)
+
+## JDBC Driver Changes
+
+* Implement `ResultSet.getStatement()`. ({issue}`4957`)
+
+## Cassandra Connector Changes
+
+* Map Cassandra `TIMESTAMP` type to Presto `TIMESTAMP(3) WITH TIME ZONE` type. ({issue}`4799`)
+
+## Hive Connector Changes
+
+* Skip stripes and row group based on Timestamp statistics for ORC files. ({issue}`1147`) 
+* Use temporary staging directory for write operation to sorted bucketed table. ({issue}`3434`)
+* Invalidate cached information on `GRANT` and `REVOKE`. ({issue}`4768`)
+* Add Parquet and RCBinary configuration properties `hive.parquet.time-zone` and `hive.rcfile.time-zone` to adjust binary timestamp 
+  values to a specific time zone. For Hive 3.1+, this should be set to UTC. The default value is the JVM default time zone, for 
+  backwards compatibility with earlier versions of Hive. ({issue}`4799`)
+* Add ORC configuration property `hive.orc.time-zone` to set the default time zone for legacy ORC files that did not declare a time 
+  zone. ({issue}`4799`)
+* Replace the `hive.time-zone` configuration property with the format specific properties. ({issue}`4799`)
+* Improve performance of partition fetching from Glue. ({issue}`4938`)
```

Comment:
This is less about performance than about cost. We used to fetch 128 partitions per API call to Glue. We now fetch 1000. The query planning performance difference with and without patch was ~<5s (out of around 3m planning time, so... negligible).

This does lower cost and help avoid running into rate-limits. Not sure how that should be conveyed in release notes.

Reply:
Although, not hitting a rate-limits does help performance.